forge/db/models/Device.js

```diff
@@ -664,9 +673,25 @@ module.exports = {
                                 }
                                 : { TeamId: teamId }),
                             ...(applicationId ? { ApplicationId: applicationId } : {})
-                        },
-                        group: ['state']
+                        }
+                    })
+
+                    const platformRbacEnabled = app.config.features.enabled('rbacApplication')
+                    const teamRbacEnabled = team.TeamType.getFeatureProperty('rbacApplication', false)
+
+                    findAll.filter((device) => {
+                        // If the device has no application or rbac not enabled, allow it through (devices without applications)
+                        if ((!platformRbacEnabled && !teamRbacEnabled) || !device.Application) {
+                            return true
+                        }
+                        const context = { applicationId: device.Application.hashid }
+                        return app.hasPermission(membership, 'device:read', context)
+                    }).forEach(device => {
```

Comment:
Minor nit - the work of this `forEach` could be done inside the `filter` to avoid iterating the list twice.